File: API/Entities/Question.cs
namespace API.Entities;

public class Question : ModeratedEntity
{
    public Guid TopicId { get; set; }

    public Topic? Topic { get; set; }

    public string Text { get; set; } = string.Empty;

    public QuestionDifficulty Difficulty { get; set; } = QuestionDifficulty.Medium;

    public QuestionType Type { get; set; } = QuestionType.Concept;

    public string Explanation { get; set; } = string.Empty;

    public string? SolutionNote { get; set; }

    public bool IsPastExamQuestion { get; set; }

    public int? ExamYear { get; set; }

    public ExamType? ExamType { get; set; }

    public ExamSession? ExamSession { get; set; }

    public string? SourceReference { get; set; }

    public string? SourceText { get; set; }

    public bool IsAiGenerated { get; set; }

    public ICollection<QuestionOption> Options { get; set; } = new List<QuestionOption>();

    public ICollection<QuizAnswer> QuizAnswers { get; set; } = new List<QuizAnswer>();
}

File: spk/Client/src/models/question.ts
import type { ContentAccessLevel, ExamSession, ExamType, QuestionDifficulty, QuestionType, ReviewStatus } from './enums'

export interface Question {
  id: string
  topicId: string
  text: string
  difficulty: QuestionDifficulty
  type: QuestionType
  explanation: string
  isPastExamQuestion: boolean
  examYear?: number | null
  examType?: ExamType | null
  examSession?: ExamSession | null
  sourceReference?: string | null
  sourceText?: string | null
  isAiGenerated: boolean
  reviewStatus: ReviewStatus
  accessLevel: ContentAccessLevel
  reviewedBy?: string | null
  reviewedAt?: string | null
  reviewComment?: string | null
  options: QuestionOption[]
}

export interface QuestionOption {
  id: string
  label: string
  text: string
  isCorrect: boolean
}

export interface CreateQuestion {
  topicId: string
  text: string
  difficulty: QuestionDifficulty
  type: QuestionType
  explanation: string
  isPastExamQuestion: boolean
  examYear?: number | null
  examType?: ExamType | null
  examSession?: ExamSession | null
  sourceReference?: string | null
  sourceText?: string | null
  isAiGenerated: boolean
  reviewStatus: ReviewStatus
  accessLevel: ContentAccessLevel
  options: CreateQuestionOption[]
}

export interface CreateQuestionOption {
  label: string
  text: string
  isCorrect: boolean
}

export type UpdateQuestion = CreateQuestion
